src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.Statements.cs

```diff
@@ -411,16 +411,23 @@ private void Statement(Block block)
             if (Context.DesignTimeMode || !isMarkup)
             {
                 // CODE owns whitespace, MARKUP owns it ONLY in DesignTimeMode.
-                if (lastWs != null)
+                if (lastWhitespace != null)
                 {
-                    Accept(lastWs);
+                    Accept(lastWhitespace);
                 }
             }
             else
             {
+                var nextSymbol = Lookahead(1);
+
                 // MARKUP owns whitespace EXCEPT in DesignTimeMode.
                 PutCurrentBack();
-                PutBack(lastWs);
+
+                // Don't putback the whitespace if it precedes a '<text>' tag.
+                if (!nextSymbol.Content.Equals(SyntaxConstants.TextTagName))
+                {
+                    PutBack(lastWhitespace);
+                }
```

Comment:
looks like this should mostly fix the problem.  but when `nextSymbol.Content.Equals(SyntaxConstants.TextTagName) && lastWhitespace != null` here (i.e. a non-existent `else if`), should we call `Accept(lastWhitespace)`?  right now nothing owns the whitespace before `<text>`